Marlin/src/inc/Conditionals_post.h

```diff
@@ -2304,83 +2332,77 @@
 //
 
 // Is an endstop plug used for extra Z endstops or the probe?
-#define IS_PROBE_PIN(A,M) (USES_Z_MIN_PROBE_PIN && Z_MIN_PROBE_PIN == A##_##M##_PIN)
+#define IS_PROBE_PIN(A,M) (HAS_Z_MIN_PROBE_PIN && Z_MIN_PROBE_PIN == A##_##M##_PIN)
 #define IS_X2_ENDSTOP(A,M) (ENABLED(X_DUAL_ENDSTOPS) && X2_USE_ENDSTOP == _##A##M##_)
 #define IS_Y2_ENDSTOP(A,M) (ENABLED(Y_DUAL_ENDSTOPS) && Y2_USE_ENDSTOP == _##A##M##_)
 #define IS_Z2_ENDSTOP(A,M) (ENABLED(Z_MULTI_ENDSTOPS) && Z2_USE_ENDSTOP == _##A##M##_)
 #define IS_Z3_ENDSTOP(A,M) (ENABLED(Z_MULTI_ENDSTOPS) && NUM_Z_STEPPER_DRIVERS >= 3 && Z3_USE_ENDSTOP == _##A##M##_)
 #define IS_Z4_ENDSTOP(A,M) (ENABLED(Z_MULTI_ENDSTOPS) && NUM_Z_STEPPER_DRIVERS >= 4 && Z4_USE_ENDSTOP == _##A##M##_)
 
-#define _HAS_STOP(A,M) (PIN_EXISTS(A##_##M) && !IS_PROBE_PIN(A,M) && !IS_X2_ENDSTOP(A,M) && !IS_Y2_ENDSTOP(A,M) && !IS_Z2_ENDSTOP(A,M) && !IS_Z3_ENDSTOP(A,M) && !IS_Z4_ENDSTOP(A,M))
-#if _HAS_STOP(X,MIN)
+#define _HAS_STOP(A,M) (PIN_EXISTS(A##_##M) && !IS_X2_ENDSTOP(A,M) && !IS_Y2_ENDSTOP(A,M) && !IS_Z2_ENDSTOP(A,M) && !IS_Z3_ENDSTOP(A,M) && !IS_Z4_ENDSTOP(A,M))
+#if   X_HOME_TO_MIN && _HAS_STOP(X,MIN)
   #define HAS_X_MIN 1
-#endif
-#if _HAS_STOP(X,MAX)
+#elif X_HOME_TO_MAX && _HAS_STOP(X,MAX)
   #define HAS_X_MAX 1
 #endif
-#if _HAS_STOP(Y,MIN)
+#if   Y_HOME_TO_MIN && _HAS_STOP(Y,MIN)
   #define HAS_Y_MIN 1
-#endif
-#if _HAS_STOP(Y,MAX)
+#elif Y_HOME_TO_MAX && _HAS_STOP(Y,MAX)
   #define HAS_Y_MAX 1
 #endif
-#if _HAS_STOP(Z,MIN)
+#if   Z_HOME_TO_MIN && _HAS_STOP(Z,MIN) && Z_MIN_PROBE_PIN != Z_MIN_PIN
```

Comment:
```
#if   Z_HOME_TO_MIN && _HAS_STOP(Z,MIN) Z_MIN_PROBE_PIN != Z_MIN_PIN
```
Results in no HAS_Z_MIN def
```
// *** CI Compatibility with old configs ***
#if ENABLED(Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN)
  #undef Z_MIN_PROBE_PIN
  #define Z_MIN_PROBE_PIN Z_MIN_PIN
#endif
```

Reply:
I expect much if this is wrong. I am documenting endstops and will have changes when I have changes.